diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-polaris.dts b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-polaris.dts
index a44d6e776..6bdb344d7 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-polaris.dts
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-polaris.dts
@@ -522,6 +522,14 @@ &pmi8998_wled {
 };
 
 &pm8998_gpios {
+	pm8998_gpio11_default: pm8998-gpio11-state {
+		pins = "gpio11";
+		function = "normal";
+		bias-pull-up;
+		drive-push-pull;
+		qcom,drive-strength = <PMIC_GPIO_STRENGTH_HIGH>;
+	};
+
 	volume_up_gpio: pm8998-gpio6-state {
 		pinconf {
 			qcom,drive-strength = <PMIC_GPIO_STRENGTH_NO>;
@@ -714,3 +722,110 @@ &wifi {
 	vdd-3.3-ch1-supply = <&vreg_l23a_3p3>;
 	status = "okay";
 };
+
+/* Camera: Sony IMX363 rear sensor (main) */
+/ {
+	cam_vio_rear: regulator-cam-vio-rear {
+		compatible = "regulator-fixed";
+		regulator-name = "cam_vio_rear";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-enable-ramp-delay = <135>;
+		enable-active-high;
+		gpio = <&tlmm 21 GPIO_ACTIVE_HIGH>;
+		vin-supply = <&vreg_s4a_1p8>;
+	};
+
+	cam_vdig_rear: regulator-cam-vdig-rear {
+		compatible = "regulator-fixed";
+		regulator-name = "cam_vdig_rear";
+		regulator-min-microvolt = <1050000>;
+		regulator-max-microvolt = <1050000>;
+		regulator-enable-ramp-delay = <135>;
+		enable-active-high;
+		gpio = <&pm8998_gpios 11 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pm8998_gpio11_default>;
+	};
+};
+
+&camss {
+	vdda-phy-supply = <&vreg_l1a_0p875>;
+	vdda-pll-supply = <&vreg_l26a_1p2>;
+	vdda-csi0-supply = <&vreg_l1a_0p875>;
+	vdda-csi1-supply = <&vreg_l1a_0p875>;
+	vdda-csi2-supply = <&vreg_l1a_0p875>;
+
+	status = "okay";
+	ports {
+		port@0 {
+			camss_csi0_ep: endpoint {
+				remote-endpoint = <&imx363_ep>;
+                            clock-lanes = <7>;
+                            data-lanes = <0 1 2 3>;
+			};
+		};
+	};
+};
+
+&cci {
+	status = "okay";
+};
+
+&cci_i2c0 {
+        clock-frequency = <400000>;
+	imx363: camera-sensor@1a {
+		compatible = "sony,imx363";
+                reset-gpios = <&tlmm 80 GPIO_ACTIVE_LOW>;
+		reg = <0x1a>;
+		status = "okay";
+
+		vana-supply = <&vreg_bob>;
+		vdig-supply = <&cam_vdig_rear>;
+		vif-supply = <&cam_vio_rear>;
+
+		clocks = <&clock_camcc CAM_CC_MCLK0_CLK>;
+		clock-names = "xvclk";
+		pinctrl-names = "default";
+		pinctrl-0 = <&cam0_default &cam_mclk0_default>;
+		clock-frequency = <24000000>;
+
+		orientation = <1>;
+		rotation = <270>;
+
+		port {
+			imx363_ep: endpoint {
+				remote-endpoint = <&camss_csi0_ep>;
+                            data-lanes = <0 1 2 3>;
+                            link-frequencies = /bits/ 64 <636000000>;
+			};
+		};
+	};
+};
+
+&tlmm {
+        cam0_default: cam0-default-state {
+            rst {
+                pins = "gpio80";
+                function = "gpio";
+                drive-strength = <2>;
+                bias-disable;
+            };
+        };
+                cam_mclk0_default: cam-mclk0-default-state {
+                pins = "gpio13";
+                function = "cam_mclk";
+                drive-strength = <2>;
+                bias-disable;
+        };
+        cam_vana_en: cam-vana-en-state {
+                gpio-hog;
+                gpios = <87 0>;
+                output-high;
+        };
+        cam_custom0_en: cam-custom0-en-state {
+                gpio-hog;
+                gpios = <102 0>;
+                output-high;
+        };
+};
