--- /tmp/qcom-ngd-ctrl-r18.c	2026-09-27 00:28:01.950126804 +0800
+++ psrc/drivers/slimbus/qcom-ngd-ctrl.c	2026-09-27 00:58:33.366116114 +0800
@@ -1020,7 +1020,8 @@
 		if (txn.msg->num_bytes == 0) {
 			int exp = 0, coef = 0;
 
-			wbuf[txn.msg->num_bytes++] = sdev->laddr;
+			wbuf[txn.msg->num_bytes++] =
+			        (port->ch.data_fmt << 5) | (sdev->laddr & 0x1f);
 			wbuf[txn.msg->num_bytes] = rt->bps >> 2 |
 						   (port->ch.aux_fmt << 6);
 
