--- /tmp/qcom-ngd-ctrl-r20.c	2026-09-27 01:32:16.342104307 +0800
+++ psrc/drivers/slimbus/qcom-ngd-ctrl.c	2026-09-27 01:32:09.206104349 +0800
@@ -1057,6 +1057,7 @@
 		wbuf[txn.msg->num_bytes++] = port->ch.id;
 	}
 
+	dev_info(&sdev->dev, "DEF_ACT_CHAN msg[%d]: %02x %02x %02x %02x %02x %02x laddr=%d\n", txn.msg->num_bytes, txn.msg->wbuf[0], txn.msg->wbuf[1], txn.msg->wbuf[2], txn.msg->wbuf[3], txn.msg->wbuf[4], txn.msg->wbuf[5], sdev->laddr);
 	dev_info(&sdev->dev, "DEF_ACT_CHAN send: tid=%d\n", txn.tid);
 	txn.mc = SLIM_USR_MC_DEF_ACT_CHAN;
 	txn.rl = txn.msg->num_bytes + 4;
